src/snowflake/snowpark/mock/mock_connection.py

```diff
@@ -64,6 +71,63 @@ def _build_target_path(stage_location: str, dest_prefix: str = "") -> str:
 
 
 class MockServerConnection:
+    class TableRegistry:
+        # Table registry. TODO: move to datastore
+        def __init__(self, conn: "MockServerConnection") -> None:
+            self.table_registry = {}
+            self.conn = conn
+
+        def get_fully_qualified_name(self, name: Union[str, Iterable[str]]) -> str:
+            def uppercase_and_enquote_if_not_quoted(string):
+                if string[0] == '"' and string[-1] == '"':
+                    return string
+                return f'"{string.upper()}"'
+
+            current_schema = self.conn._get_current_parameter("schema")
```

Comment:
Can this be `None`?

Reply:
In the current `mock_connection.py`, this is hardcoded to have a value. For a real connection if there is no current schema, it should raise an exception when executing the query.